src/device/mod.rs

```diff
@@ -2027,6 +2045,19 @@ impl TaskRuntime for Runtime {
                 let node = self.peer_to_node(&mesh_event.peer, Some(mesh_event.state), mesh_event.link_state).await;
 
                 if let Some(node) = node {
+                    if mesh_event.state != PeerState::Connecting && node.path == PathType::Relay {
+                        let event = AnalyticsEvent{
+                            public_key: mesh_event.peer.public_key,
+                            dual_ip_addresses: mesh_event.peer.get_dual_ip_addresses(),
+                            tx_bytes: mesh_event.peer.tx_bytes.unwrap_or_default(),
+                            rx_bytes: mesh_event.peer.rx_bytes.unwrap_or_default(),
+                            peer_state: mesh_event.state,
+                            timestamp: Instant::now(),
+                        };
```

Comment:
The only difference the constructor would make is that it could automatically fill `timestamp`. Not sure that this is worth it. Let me know what you think.

Reply:
I think that moving this logic to some `From` trait and just using it here would be nice

Reply:
I don't think this is good idea - this is not a lossless conversion. See: https://doc.rust-lang.org/std/convert/trait.From.html#when-to-implement-from